core/src/test/java/org/apache/iceberg/TestOverwrite.java

```diff
@@ -312,4 +313,154 @@ public void testValidatedOverwriteWithAppendSuccess() {
     Assert.assertEquals(
         "Should not create a new snapshot", baseId, latestSnapshot(base, branch).snapshotId());
   }
+
+  @Test
+  public void testOverwriteWithEmptyTableAppendManifest() throws IOException {
+    Assert.assertEquals("Table should start empty", 0, listManifestFiles().size());
+
+    TableMetadata base = readMetadata();
+    Assert.assertNull("Should not have a current snapshot", latestSnapshot(base, branch));
```

Comment:
Please use `AssertJ` framework.
https://iceberg.apache.org/contribute/#assertj

Reply:
I was following class template. I will update for all the tests.